### ISSUE TITLE
Part of time period

### ISSUE THREAD

**Issue by vladdeSV** [jotaen/klog#42]
This is more of a question than a suggestion.

At my workplace I need to keep track of two things:
- the time period I am at the workplace (when I get there, and when I leave)
- the time I spend with specific projects

Usually, I do a multitude of things for my department. But sometimes I get assigned to a project where we bill for the time we spent on that project.

What I want to do is to write down an entry when I get to work and when I leave. Then I'd like to make an entry for the time spend with the project, which would be a "part" of that time. At the end of the month I'd like to summarize the total of hours spent with that project.

The only way I think I could do this as of right now would be:

```klog
2021-03-26
  08:00 - 17:00
  -1h lunch
  4h20m #foo
  -4h20m
```

This will give that I worked 8 hours that day, and when I summarize `#foo`, it would give `4h20m`. Is there an best practice to do what I want?

(If this was to be some additional feature to the language, I have no idea how that would work.)

**Comment by acikek** [comment]
Maybe there could be an "ignore total" feature for a tracker, so your above code would be:

```klg
2021-03-26
  08:00 - 17:00
  -1h lunch
  *4h20m #foo
```

**Comment by vladdeSV** [comment]
Ah, maybe some symbol before a duration could be something! An alternative might be the exclamation sign `!`, as it is used in the should-total as well.

If something like was to be implemented, I think that would be a non-intrusive feature.

**Comment by acikek** [comment]
@jotaen What are your thoughts on this? I think the general "time tracker decorators" (another being #41) concept would be quite cool.

**Comment by jotaen** [comment]
This would be kind of a breakdown of times, wouldn’t it? Like: I did something for 5h; 3h of which this and 2h of which that.

As far as I see this only makes sense in conjunction with tags, right? At least when it comes to the evaluation? I’m not sure, though, how this could be conceptualised transparently, because if you say the `*` (or whatever) ignores the time entry, then you still want to count it when evaluating the tags.

Maybe related inspiration, (h)ledger (a plain-text accounting format/tool) has a feature called [“virtual postings”](https://hledger.org/hledger.html#virtual-postings) where you put entries in parenthesis and then they are only taken into consideration under specific circumstances. It’s useful on the one hand, but it also introduces some hidden complexity.

I think such a feature could make a powerful addition, but I feel it must be carefully designed.

**Comment by vladdeSV** [comment]
> but I feel it must be carefully designed.

Exactly my thoughts.

@jotaen: Should another issue be created for the idea proposed in this thread? My initial though was some sort of subset of a time period, and the idea proposed here is a conditional entry.

**Comment by jotaen** [comment]
> Should another issue be created for the idea proposed in this thread?

You mean the “ignoring”? As far as I understood this was also be supposed a solution to the use-case you described, not sure currently how useful it is on its own. On the other hand there also was a discussion around [point-in-time](https://github.com/jotaen/klog/issues/30) entries, so maybe this is all sort-of related… 🤔

**Comment by vladdeSV** [comment]
Hm. I do not really know what is best for *klog*, so I am conveniently going to push that decision onto you ;)

The issue is very much related to me writing down everything manually, which I also briefly mention in a comment in #3

**Comment by acikek** [comment]
> I am a bit worried about feature-creep

As one who has contributed to this, first of all, I am sorry - but yes, I have felt this too. I think that my idea and this one could be implemented carefully and elegantly, but at what point do we draw the line on adding new, incredibly specific use-case features? Is klog really the format for this purpose?

Time is complicated, and how people use it even more so.

**Comment by jotaen** [comment]
Yeah, I feel the same. I also like the current set of functionalities and perceive it as relatively self-contained and complete. I’m open to new ideas and enhancements, but keeping klog simple and straightforward is the primary goal. The line-drawing is tricky in any event, as you mentioned the use-cases are vastly different and there is no one-size-fits-all solution. The good thing is there are approximately 1283992 other time tracking tools, so people have plenty of choice.

I cleaned up the issue labels a bit, to make it clearer which ideas are still tentative and which are more concrete. Just as fyi, I’m currently in favour of adding [multiline entry summaries](https://github.com/jotaen/klog/issues/38) and [negative time ranges](https://github.com/jotaen/klog/issues/43), because they seem to make sense in itself and are strictly optional (i.e. not affecting or interferring with existing functionality). I won’t rush this, though, and I also wanted to finalise the v1.0 file-spec before adding anything new to the file format.
